frontend/ui/widget/textboxwidget.lua

```diff
@@ -1168,6 +1169,28 @@ function TextBoxWidget:update(scheduled_update)
     self.scheduled_update = nil
 end
 
+function TextBoxWidget:setText(text)
+    if text == self.text then
+        return
+    end
+
+    self.text = text
+
+    if self.use_xtext then
+        self:_measureWithXText()
+    else
+        self:_evalCharWidthList()
+    end
+    self:_splitToLines()
+
+    self:update()
```

Comment:
Not sure I would trust this. Seems a few more things may be done in :init() like dealing with height, that doing this quick fake-init() might mess ?